### PR TITLE
feat(business): Add power feature icons 

### PR DESCRIPTION
**Pull request by denamwangi**
This will make room for entry points in the main sidebar and settings sidebar as well as at feature gate points that currently have the lock icon.

Related to https://github.com/getsentry/getsentry/pull/2624

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/organizationAuth/providerItem.jsx [hunk 1]
```diff
@@ -125,7 +125,7 @@ const LockedFeature = ({provider, features, className, ...props}) => (
       />
     }
   >
-    <Tag icon="icon-lock">disabled</Tag>
+    <Tag icon="icon-power-features">disabled</Tag>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/organizationAuth/providerItem.jsx]: Let's leave this as a lock. This will only show up for on premis users.

File: src/sentry/static/sentry/app/components/sidebar/index.jsx [hunk 2]
```diff
@@ -266,17 +267,26 @@ class Sidebar extends React.Component {
...
-                      this.navigateWithGlobalSelection(
-                        `/organizations/${organization.slug}/events/`,
-                        evt
-                      )}
-                    icon={<InlineSvg src="icon-stack" />}
-                    label={t('Events')}
-                    to={`/organizations/${organization.slug}/events/`}
-                  />
+                  <StyledItem>
+                    <SidebarItem
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/index.jsx]: So, what I would actually like to do here is make `SidebarItem` use hooks inside of itself to determine what should be rendered as part of the item.

Right now this explicitly adds hooks into specific sidebar items that we know we want icons next to in getsentry. But that logic, knowing which sidebar items to put icons next to, is logic that only getsentry should be aware of. This is in the same spirit of having a generic name for the hook, we want to try and avoid adding in getsentry specific logic / functionality into sentry.

So what I think we could do is setup a hook that overrides the sidebar label, allowing customization of the label, which is more general functionality

https://github.com/getsentry/sentry/blob/0f09b27c2012f866e75aa59a56db1c86bad6f643/src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx#L91-L93

So here we can plug a hook in here and call it something like `sidebar:item-label`, which allows us to override the item label like so

```jsx
<Hook name="sidebar:item-label">
    {({hooks}) =>  hooks.map(h => h({
        organization,
        id: this.props.id // This is new, SidebarItems will have to specify ids.        
        label: (
                <SidebarItemLabel>
                    <TextOverflow>{label}</TextOverflow>
                </SidebarItemLabel>
            ),
        }))}
</Hook>
```

Or well, something close to that. This won't work 100% since if there's no hook defined it will return null, which means we'll just endup with empty labels on sentry.

Then in getsentry you can add a hook that will look at the label id and determine if it should add the power icon for it.

/cc @billyvg for thoughts

File: src/sentry/static/sentry/app/components/sidebar/index.jsx [hunk 3]
```diff
@@ -412,6 +439,11 @@ class Sidebar extends React.Component {
         {hasOrganization && (
           <SidebarSectionGroup>
             <SidebarSection>
+              {HookStore.get('sidebar:bottom-items').length &&
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/index.jsx]: You can just use the `Hook` component here.

File: src/sentry/static/sentry/app/components/sidebar/index.jsx [hunk 4]
```diff
@@ -566,3 +598,8 @@ const SidebarCollapseItem = styled(SidebarItem)`
     display: none;
   }
 `;
+const StyledItem = styled.div`
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/index.jsx]: I think this isn't needed here?

File: src/sentry/static/sentry/app/stores/hookStore.jsx [hunk 5]
```diff
@@ -22,6 +22,8 @@ const validHookNames = new Set([
   'utils:onboarding-survey-url',
   'component:org-auth-view',
   'component:org-members-view',
+  'sidebar:bottom-items',
+  'sidebar:secondary-icon',
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/stores/hookStore.jsx]: Lets sort these up with the other sidebar items. I think each group here is sorted.

File: src/sentry/static/sentry/app/views/settings/components/settingsNavigationGroup.jsx [hunk 6]
```diff
@@ -48,13 +55,21 @@ export default class NavigationGroup extends React.Component {
...
+            <StyledItem key={title}>
+              <SettingsNavItem
+                key={title}
+                to={to}
+                label={title}
+                index={index}
+                badge={badgeResult}
+              />
+              <Hook
+                name="sidebar:secondary-icon"
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/components/settingsNavigationGroup.jsx]: Same idea here as above for hooking into nav items, but lets call it `settings-nav:item-label`

File: src/sentry/static/sentry/app/components/hovercard.jsx [hunk 7]
```diff
@@ -81,7 +82,7 @@ class Hovercard extends React.Component {
   }
 
   render() {
-    const {containerClassName, className, header, body} = this.props;
+    const {containerClassName, className, header, body, bodyClassName} = this.props;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/hovercard.jsx]: Can you move this parameter to the front to go with the other class parameters.

File: src/sentry/static/sentry/app/components/hovercard.jsx [hunk 8]
```diff
@@ -15,6 +15,7 @@ class Hovercard extends React.Component {
     containerClassName: PropTypes.string,
     header: PropTypes.node,
     body: PropTypes.node,
+    bodyClassName: PropTypes.string,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/hovercard.jsx]: If you want to get this in asap this can be a separate PR (for the hovercard stuff)

File: src/sentry/static/sentry/app/components/sidebar/index.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/sidebar/index.jsx]: Yeah sounds good - would `HookOrDefault` help here?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/sidebar/index.jsx]: Didn't know about `HookOrDefault`. 👍

File: src/sentry/static/sentry/app/components/sidebar/index.jsx (hunk 3 above)
Comment (denamwangi) [reply src/sentry/static/sentry/app/components/sidebar/index.jsx]: oo yeah so I started out using `Hook` but once the component mounts updates to props don't propagate for some reason I need to look into which affects the collapsed state :/

File: src/sentry/static/sentry/app/components/sidebar/index.jsx (hunk 2 above)
Comment (denamwangi) [reply src/sentry/static/sentry/app/components/sidebar/index.jsx]: Yeah hook or default works well here for now but def agree that we could refactor `Hook` with the `default`

File: src/sentry/static/sentry/app/components/sidebar/index.jsx (hunk 3 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/sidebar/index.jsx]: /cc @billyvg is this a known issue with the `Hook`?

File: src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx [hunk 9]
```diff
@@ -69,6 +71,10 @@ class SidebarItem extends React.Component {
 
     const isTop = orientation === 'top';
     const placement = isTop ? 'bottom' : 'right';
+    const TextOverflowHook = HookOrDefault({
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx]: IMO it should be a component outside of the `TextOverflow`, since we don't specifically want to override the overflow wrapper (we want to keep that). Instead I would do something like this:

```js
const LabelHook = HookOrDefault({
    hookName: 'sidebar:item-label',
    defaultComponent: React.Fragment,
});
```

And then 

```jsx
<LabelHook>
    <TextOverflow>{label}</TextOverflow>
</LabelHook>
```

File: src/sentry/static/sentry/app/views/settings/components/settingsNavItem.jsx [hunk 10]
```diff
@@ -4,20 +4,28 @@ import React from 'react';
...
 
   render() {
-    const {badge, label, index, ...props} = this.props;
+    const {badge, label, index, id, ...props} = this.props;
 
     let renderedBadge = '';
 
+    const LabelHook = HookOrDefault({
+      hookName: 'sidebar:item-label',
+      defaultComponent: TextOverflow,
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/components/settingsNavItem.jsx]: Same as above.

File: src/sentry/static/sentry/app/views/settings/components/settingsNavItem.jsx [hunk 11]
```diff
@@ -46,6 +56,7 @@ const StyledNavItem = styled(Link)`
   font-size: 14px;
   line-height: 30px;
   position: relative;
+  width: 100%;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/components/settingsNavItem.jsx]: Why doers it need this, shouldn't display block already ensure that?

File: src/sentry/static/sentry/app/components/sidebar/index.jsx [hunk 12]
```diff
@@ -276,6 +277,7 @@ class Sidebar extends React.Component {
                     icon={<InlineSvg src="icon-stack" />}
                     label={t('Events')}
                     to={`/organizations/${organization.slug}/events/`}
+                    id="events"
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/index.jsx]: Did we add `id` to all the sidebar items?

File: src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx (hunk 9 above)
Comment (denamwangi) [reply src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx]: That's cleaner- updated!

File: src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx [hunk 13]
```diff
@@ -95,7 +95,11 @@ class SidebarItem extends React.Component {
             {!collapsed &&
               !isTop && (
                 <SidebarItemLabel>
-                  {<TextOverflowHook id={this.props.id}>{label}</TextOverflowHook>}
+                  {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx]: I don't think you need the bace here do you?

File: src/sentry/static/sentry/app/components/sidebar/index.jsx [hunk 14]
```diff
@@ -411,6 +426,11 @@ class Sidebar extends React.Component {
         {hasOrganization && (
           <SidebarSectionGroup>
             <SidebarSection>
+              {HookStore.get('sidebar:bottom-items').length > 0 &&
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/index.jsx]: Gonna try and look into why we can't use the Hook component here 😢

File: src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx [hunk 15]
```diff
@@ -69,6 +71,12 @@ class SidebarItem extends React.Component {
 
     const isTop = orientation === 'top';
     const placement = isTop ? 'bottom' : 'right';
+    const LabelHook = HookOrDefault({
+      hookName: 'sidebar:item-label',
+      defaultComponent: ({children}) => {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/sidebar/sidebarItem.jsx]: formatting nit, not important or worth changing, but you could write this minorly shorter by not using braces for a function body

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/components/sidebar/index.jsx (hunk 14 above)
Comment (denamwangi) [reply src/sentry/static/sentry/app/components/sidebar/index.jsx]: That would be great - it's definitely cleaner to use `Hook`. It hasn't come up for anything else i've used it for so I was surprised by the behaviour. Once we get that fixed up and refactor to add `default` to it we can swap it in
